krishnanlab/simplex
# Enable API on frontend

- update all packages to latest versions
- reorder api funcs to match exact order in api "docs" (#3)
- rename some api funcs and change some request methods to match api
- change article `revision` field to be `Id` type instead of number
- sort `revisions` api response by date (oldest first)
- author id now `null` when anonymous
- add debug logging for all requests
- update mock data to reflect change to article `revision` field
- make animated icon to indicate user has to select word to get help
- normalize comma-separated array input field to lower case
- change editor tooltip to activate on select instead of click because click was getting annoying when trying to type
- normalize editor highlighting to lower case
- add password meter component using `zxcvbn` and incorporate into sign up and change password pages
- rename "loggedIn" global state var to more appropriate "currentUser"
- allow tooltips to accept any react node instead of just string
- add scroll behavior to notifications, and allow array of items to be shown as error message
- render wikipedia definition as html instead of plain text because sometimes it seems to include html
- add `current-user` global event, and use `useEvent` to wait for it when trying to direct (e.g., on account page, wait for initial app-load current user check to finish before redirecting to login page if user not logged in)
- comment out MSW (mocked api responses)
- add error messages from back end into all queries, instead of just generic message e.g. "error loading article"
- save/restore editable article state (e.g. text typed into box) to/from local storage in case user accidentally closes app
- more comprehensively remove react-query query caches
- add text balancer library for hero text (can be removed in a few years once the `text-wrap` css proposal lands in all browsers)
- move anonymous articles section to homepage (previously on my-articles page, which is only for logged in users)
- "remember me" unchecked by default to avoid legal issues
- add "remember me" checkbox to signup too
- show "anonymous" if author field of article is null
- add try catch around analyze call
- add ? tooltip next to newsletter with "promise" note
- fix get/set storage function bugs
- rework string util funcs for tokenizing
- add util func to pretty-format django rest frame work standard errors to show to user

closes #10
